crates/pedersen/src/curve_consts_gen.rs

```diff
@@ -0,0 +1,102 @@
+use crate::curve::*;
+use std::fs;
+
+#[allow(dead_code)]
+pub fn generate_consts(path: &str, bits: u32) {
```

Comment:
If we'd made this into

```suggestion
pub fn generate_consts(path: &str, bits: u32) -> Result<(), std::io::Error> {
```

Then we could use `writeln!` (maybe `use std::fmt::Write;` was required) to push stuff into the String, easing up a bit on the `buf.push_str(format!(....).as_str());` which would become `writeln!(buf, "something formatted {}", args)?;` and turn `buf.push_str("....\n")` into a `writeln!(buf, "...")?;`.

Reply:
Agreed :-)